sdks/go/pkg/beam/core/graph/edge.go

```diff
@@ -202,16 +202,16 @@ func NewCoGBK(g *Graph, s *Scope, ns []*Node) (*MultiEdge, error) {
 	for i := 1; i < len(ns); i++ {
 		n := ns[i]
 		if !typex.IsKV(n.Type()) {
-			return nil, fmt.Errorf("input type must be KV: %v", n)
+			return nil, fmt.Errorf("creating new CoGBK in scope %v: input type must be KV: %v", s, n)
```

Comment:
Mostly just a thought, No Action Required.

Given an offline discussion, you've probably thought about this, but note how the context is identical for each of these errors. Having a dedicated error type could reduce the repetition of the context. The trick is embedding and ensuring there's data for the subsequent errors (eg with a more general approach with format strings and []interface{} args.)

Reply:
Ack. Yeah, I considered reducing the repetition by creating an error type, but I was already thinking about creating an error package for Beam and didn't really wanna do anything too complex in case it wouldn't work with the new errors. I'll add a TODO with a bug though so I don't forget to improve this later.